File: golang-sdk/common/hash.go
package common

import (
	"crypto/hmac"
	"crypto/sha256"
	"hash"
)

func Sha256Hash(message []byte) ([]byte, error) {
	return calculateHash(sha256.New(), message)
}

func HmacSha256(key, message []byte) ([]byte, error) {
	h := hmac.New(sha256.New, key)
	return calculateHash(h, message)
}

func calculateHash(h hash.Hash, data []byte) ([]byte, error) {
	_, err := h.Write(data)
	if err != nil {
		return nil, err
	}
	return h.Sum(nil), nil
}

File: swift-sdk/Sources/WikibrokerOpenapiSdk/Common/Hash.swift
import CryptoKit
import Foundation

public func sha256Hash(message: Data) -> Data {
    return Data(SHA256.hash(data: message))
}

public func hmacSha256(key: Data, message: Data) -> Data {
    let symmetricKey = SymmetricKey(data: key)
    let authCode = HMAC<SHA256>.authenticationCode(
        for: message,
        using: symmetricKey
    )
    return Data(authCode)
}
